packages/api/tests/mock.factories.js

```diff
@@ -1137,11 +1137,30 @@ function fakeProduct(defaultData = {}) {
     name: faker.lorem.words(2),
     supplier: faker.lorem.words(3),
     on_permitted_substances_list: faker.helpers.arrayElement(['YES', 'NO', 'NOT_SURE']),
-    type: faker.helpers.arrayElement(['soil_amendment_task', 'pest_control_task', 'cleaning_task']),
+    // For soil_amendment_task use soil_amendment_productFactory
+    type: faker.helpers.arrayElement(['pest_control_task', 'cleaning_task']),
     ...defaultData,
   };
 }
 
+async function soil_amendment_productFactory({ promisedProduct = productFactory() } = {}) {
+  const [{ product_id, type }] = await promisedProduct;
+  const productDetails = fakeProductDetails(type);
+  return knex('soil_amendment_product')
+    .insert({ product_id, ...productDetails })
+    .returning('*');
+}
+
+function fakeProductDetails(type, defaultData = {}) {
```

Comment:
It's hard to imagine how this function will be utilized... you're thinking of the possibility of updating `...defaultData`(line 1157), right? 😂

Reply:
No product details (soil_amendment_product) are required but the shape of the object is. `{}` lol

If you want to test npk on backend or something you could update this to have default values or pass them to override.